backend/src/common/helpers/create-witelisted-routes-helper.ts

```diff
@@ -0,0 +1,12 @@
+import { type ServerAppApi } from '~/common/server-application/types/types.js';
+
+import { extractApiPath } from './extract-api-path-helper.js';
+
+const createWhitelistedRoutes = (apis: ServerAppApi[]): string[] => {
+    const filteredRoutes = apis.flatMap((api) =>
+        api.routes.filter((element) => !element.isPublic),
+    );
+    return filteredRoutes.map((item) => extractApiPath(item.path)) as string[];
+};
```

Comment:
you don't need `extractApiPath` here, item.path should be in same format as `request.routerOptions.user`, but double-check please